core/includes/form.inc

```diff
@@ -3168,6 +3168,151 @@ function map_month($month) {
+  $attributes['value'] = $element['#value'];
+  $attributes['name'] = $element['#name'];
+  $attributes['step'] = 1;
+
+  $output = '<input ' . backdrop_attributes($attributes) . ' />';
+  return $output;
+}
+
+/**
+ * Returns html5 markup for a combined date and time form element.
+ */
+function theme_html_datetime($variables) {
+  $element = $variables['element'];
+  $attributes = array();
+  if (isset($element['#id'])) {
+    $attributes['id'] = $element['#id'];
+  }
+  if (!empty($element['#attributes']['class'])) {
+    $attributes['class'] = (array) $element['#attributes']['class'];
+  }
+  $attributes['class'][] = 'container-inline';
+  return '<div' . backdrop_attributes($attributes) . '>' . backdrop_render_children($element) . '</div>';
+}
+
+/**
+ * Combine html_date and html_time form elements.
+ */
+function form_process_html_datetime($element) {
+  $value = $element['#value'];
+  // Conversion to a timestamp has happened, possibly in a preview.
+  // Get it back into an array for the form.
+  if (is_numeric($value) && (int) $value == $value) {
+    $value = array(
+      'date' => format_date($value, 'custom', DATE_FORMAT_DATE),
+      'time' => format_date($value, 'custom', DATE_FORMAT_TIME),
+    );
+  }
+  // Catch possible problems.
+  if (!is_array($value) || !isset($value['date']) || !isset($value['time'])) {
+    $value = array(
+      'date' => '',
+      'time' => '',
+    );
+  }
+  $element['#tree'] = TRUE;
+  $element['date'] = array(
+    '#type' => 'html_date',
+    '#default_value' => $value['date'],
+  );
+  $attributes = $element['#attributes'];
+  if (isset($attributes['date'])) {
+    $element['date']['#attributes'] = $attributes['date'];
+  }
+
+  $element['time'] = array(
+    '#type' => 'html_time',
+    '#default_value' => $value['time'],
+  );
+  if (isset($attributes['time'])) {
+    $element['time']['#attributes'] = $attributes['time'];
+  }
+
+  return $element;
+}
+
+/**
+ * Validate a date string.
+ */
+function html_date_validate($element, &$form_state) {
+  $date = $element['#value'];
+  if (!empty($date)) {
+    $valid_date = DateTime::createFromFormat(DATE_FORMAT_DATE, $date);
+    if (!$valid_date) {
+      form_error($element, t('%date is not a valid date.', array('%date' => $date)));
+    }
+  }
+}
+
+/**
+ * Validate a time string.
+ */
+function html_time_validate($element, &$form_state) {
+  $time = $element['#value'];
+  if (!empty($time)) {
+    // Pad time if seconds were omitted regardless of the "step" attribute.
+    // Some browsers do so, but our date pattern contains seconds.
+    if (preg_match('/^\d{2}:\d{2}$/', $time)) {
+      $time .= ':00';
+    }
+    $valid_time = DateTime::createFromFormat(DATE_FORMAT_TIME, $time);
+    if (!$valid_time) {
+      form_error($element, t('%time is not a valid time.', array('%time' => $time)));
+    }
+  }
+}
+
+/**
+ * Validate a combined date and time array.
+ */
+function html_datetime_validate($element, &$form_state) {
```

Comment:
Currently when submitting a scheduled or authored date with a blank time it will default to 12:00. But with PR throws a validation error if time is blank. We may need to have those specific fields have an explicit default for time now or something like that.

Reply:
I agree, we shouldn't change current behavior (time gets set to 00:00 if not set at all). 
It's easy to set a default value, though. If I only could remember, why I did it that way back in September.